backend/src/test/java/edu/alexu/cse/dripmeup/UserServiceTest.java

```diff
@@ -31,35 +34,45 @@ void setUp() {
     void testLogin_Successful() {
         UserEntity mockUser = new UserEntity();
         mockUser.setEmail("test@example.com");
-        mockUser.setPassword("password123");
+        String rawPassword = "password123";
+        String encodedPassword = passwordEncoder.encode(rawPassword);
+        mockUser.setPassword(encodedPassword);
 
         when(userRepository.findByEmail("test@example.com")).thenReturn(mockUser);
+        when(passwordEncoder.matches(rawPassword, encodedPassword)).thenReturn(true);
```

Comment:
This line will invalidate the test, since the purpose of the test is to check that the method could find if raw Password matches with encoded Password.